include/dr/details/ranges_shim.hpp

```diff
@@ -0,0 +1,27 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+// TODO: use libstdc++ 13.0 or greater if available.
+
+// #define DR_USE_STD_RANGES
```

Comment:
I haven't figured out the correct macros yet, but we should be able to automatically include standard ranges if we detect that the standard library is recent enough.

Reply:
I would like to be able to force it from CMAKE, which would require `ifdef` control. We can also support auto selection.

Reply:
Sounds good, so we'll have maybe a `DR_USE_STD_RANGES` and a `DR_USE_RANGE_V3` that can be set, and if neither of those are set we'll select automatically?

Reply:
Having auto selection only (no flag setting in cmake) will limit test cases we need to support (e.g. gcc-13 with v3-ranges will not be needed to be tested).

Reply:
We can decide not to support some combinations, like gcc-13 + range-v3, but I still want to be able to configure it that way for isolating issues.